src/components/VirtualKeyboard.vue

```diff
@@ -16,27 +18,42 @@ const props = defineProps<{
   heldNotes: Map<number, number>
 }>()
 
-type VirtualKey = {
+export type VirtualKey = {
   x: number
   y: number
   index: number
-  color: string
+  color: string,
+  frequency: number,
+  cent: number,
+  ratio: number,
+  label: string
 }
 
+const scale = useScaleStore();
```

Comment:
I've tried to keep the components store independent when reasonable. Take everything you need at the view level and pass in as props.